src/main/java/io/github/pylonmc/pylon/base/content/building/Immobilizer.java

```diff
@@ -47,7 +48,7 @@ public Item(@NotNull ItemStack stack) {
         }
     }
 
-    public static Set<Player> frozenPlayers = new HashSet<>();
+    public static Set<Player> frozenPlayers = ConcurrentHashMap.newKeySet();
```

Comment:
Don't need it concurrent if its run sync

Reply:
That's my point: now that it's sync, we dont need this set to be concurrent as it will only ever be accessed from one thread